src/coreclr/vm/ceeload.cpp

```diff
@@ -1405,14 +1407,21 @@ void Module::FreeClassTables()
             {
                 TypeHandle th = pEntry->GetTypeHandle();
 
+                // Type desc is handled differently
```

Comment:
```suggestion
                // Array EEClass doesn't need notification and there is no work for Destruct()
```
That array comment below was describing why the type desc check is here

Reply:
awesome, thanks for the suggestion! I didn't realize that this was about array classes